README.md

```diff
@@ -444,3 +514,187 @@ steps:
+  fail-fast: false  # don't stop jobs if one fails
+    steps:
+      - uses: coq-community/docker-coq-action@v1
+        with:
+          opam_file: 'coq-demo.opam'
+          custom_image: ${{ matrix.image }}
+          before_script: |
+            startGroup "Workaround permission issue"
+              sudo chown -R coq:coq .
+            endGroup
+          script: |
+            startGroup "Build project"
+              coq_makefile -f _CoqProject -o Makefile
+              make -j2
+            endGroup
+          after_script: |
+            set -o pipefail  # recommended if the script uses pipes
+
+            startGroup "Build artifacts"
+              mkdir -v -p artifacts
+              opam list > artifacts/opam_list.txt
+              make test 2>&1 | tee artifacts/make_test.txt
+            endGroup
+          uninstall: ''
+      - name: Revert permissions
+        # to avoid a warning at cleanup time
+        if: ${{ always() }}
+        run: sudo chown -R 1001:116 .
+      - uses: actions/upload-artifact@v2
+        with:
+          name: example-artifact
+          path: artifacts/
+          if-no-files-found: error  # 'warn' or 'ignore' are also available, defaults to `warn`
+          retention-days: 8
+```
+
+### Install Debian packages
+
+If you use `docker-coq-action` with a
+[Docker-Coq](https://github.com/coq-community/docker-coq) image (the
+default when the [`custom_image`](#custom_image) field is omitted),
+the image is based on Debian stable and the container user
+(UID=GID=1000) has `sudo` rights, so you can rely on `apt-get` to
+install additional Debian packages.
+
+This use case is illustrated by the following job that installs
+the `emacs` and `tree` packages:
+
+```yaml
+runs-on: ubuntu-latest
+strategy:
+  matrix:
+    image:
+      - 'coqorg/coq:dev'
+  fail-fast: false  # don't stop jobs if one fails
+steps:
+  - uses: actions/checkout@v2
+  - uses: coq-community/docker-coq-action@v1
+    with:
+      opam_file: 'folder/coq-proj.opam'
+      custom_image: ${{ matrix.image }}
+      before_script: |
+        startGroup "Install APT dependencies"
+          cat /etc/os-release  # Print the Debian OS version
+          sudo apt-get update -y -q  # Mandatory
+          sudo DEBIAN_FRONTEND=noninteractive apt-get install -y -q --no-install-recommends \
+            emacs \
+            tree  # for instance (in alphabetical order)
+        endGroup
+      after_script: |
+        startGroup "Post-test"
+          emacs --version
+          tree
+        endGroup
+```
+
+### Verbose output and variable leaking
+
+The code run in the `docker-coq-action` container relies on the
+following invocation to display a customized prompt before each
+command:
+
+```bash
+export PS4='+ \e[33;1m($0 @ line $LINENO) \$\e[0m '; set -ex
+```
+
+As a result, due to the `set -x` option, the value of each variable is
+exposed in the log.
+
+For example, the script
+
+```bash
+startGroup "Toy example"
+  echo "ex_var=$ex_var"
```

Comment:
IMHO the `echo` example is not the most pedagogical because it is expected that the value will be printed, and therefore, the reader may miss the point that it is printed *twice* (in the command call and in the output). Choosing some command which is not supposed to print any output would be clearer, I think.